src/lib/getRunnerOSVersion.js

```diff
@@ -0,0 +1,85 @@
+const fs = require('node:fs')
+const core = require('@actions/core')
+
+const IMAGE_OS_TO_RUNNER_OS = {
+  ubuntu18: 'ubuntu-18.04',
+  ubuntu20: 'ubuntu-20.04',
+  ubuntu22: 'ubuntu-22.04',
+  ubuntu24: 'ubuntu-24.04',
+  win19: 'windows-2019',
+  win22: 'windows-2022',
+}
+
+function getRunnerOSVersion(imageOS, readFile = fs.readFileSync) {
+  core.startGroup('Get runner OS version')
+
+  core.info(`ImageOS is set to '${imageOS}'`)
+
+  if (imageOS === 'Linux') {
```

Comment:
Would it be better to just look at `/etc/os-release` existing than this? Reason for it is we seem to be moving from one constraint to the other. If we want to be more generic we can base ourselves on the standard (`/etc/os-release`) and not the results we're looking at today.